packages/@adobe/spectrum-css-temp/components/inlinealert/index.css

```diff
@@ -0,0 +1,117 @@
+  --spectrum-inlinealert-heading-font-style: var(--spectrum-global-font-style-regular);
+  --spectrum-inlinealert-heading-font-size: var(--spectrum-global-dimension-font-size-100);
+  --spectrum-inlinealert-heading-line-height: var(--spectrum-alias-heading-text-line-height);
+
+  --spectrum-inlinealert-content-font-weight: var(--spectrum-alias-body-text-font-weight);
+  --spectrum-inlinealert-content-font-style: var(--spectrum-global-font-style-regular);
+  --spectrum-inlinealert-content-font-size: var(--spectrum-global-dimension-font-size-100);
+  --spectrum-inlinealert-content-line-height: var(--spectrum-alias-line-height-body);
+
+  /* Size*/
+  --spectrum-inlinealert-border-width: var(--spectrum-alias-border-size-thick);
+  --spectrum-inlinealert-border-radius: var(--spectrum-border-radius);
+  --spectrum-inlinealert-icon-size: var(--spectrum-global-dimension-size-225);
+  --spectrum-inlinealert-min-inline-size: 240px;
+  --spectrum-inlinealert-header-min-block-size: var(--spectrum-global-dimension-size-250);
+
+  /* Spacing*/
+  --spectrum-inlinealert-spacing-edge-to-text: var(--spectrum-global-dimension-static-size-300);
+  --spectrum-inlinealert-spacing-header-to-icon: var(--spectrum-global-dimension-static-size-300);
+  --spectrum-inlinealert-spacing-header-content-button: var(--spectrum-global-dimension-static-size-200);
+
+  /* Color */
+  --spectrum-inlinealert-background-color: var(--spectrum-alias-background-color-default);
+  --spectrum-inlinealert-border-and-icon-color: var(--spectrum-gray-visual-color);
+  --spectrum-inlinealert-header-color: var(--spectrum-alias-heading-text-color);
+  --spectrum-inlinealert-content-color: var(--spectrum-alias-text-color);
+  --spectrum-inlinealert-border-and-icon-color-info: var(--spectrum-informative-visual-color);
+  --spectrum-inlinealert-border-and-icon-color-positive: var(--spectrum-positive-visual-color);
+  --spectrum-inlinealert-border-and-icon-color-notice: var(--spectrum-notice-visual-color);
+  --spectrum-inlinealert-border-and-icon-color-negative: var(--spectrum-negative-visual-color);
+}
+
+.spectrum-InLineAlert {
+  position: relative;
+
+  display: inline-block;
+  box-sizing: border-box;
+  min-inline-size: var(--spectrum-inlinealert-min-inline-size);
+
+  padding-block: var(--spectrum-inlinealert-spacing-edge-to-text);
+  padding-inline: var(--spectrum-inlinealert-spacing-edge-to-text);
+
+  border-block-width: var(--spectrum-inlinealert-border-width);
+  border-inline-width: var(--spectrum-inlinealert-border-width);
+  border-style: solid;
+  border-radius: var(--spectrum-inlinealert-border-radius);
+}
+
+.spectrum-InLineAlert .spectrum-InLineAlert-grid {
+  display: grid;
+  grid-template-columns: 1fr auto;
+  grid-template-rows: auto auto auto;
+  column-gap: var(--spectrum-inlinealert-spacing-header-to-icon);
+  grid-template-areas:
+    "header      icon       "
+    "content     content    ";
```

Comment:
this combination means that the header will always stop "column-gap" away from the containing elements 'inline end'

we'll need to make use of `useHasChild` to determine if we have an icon, then apply a class either to the grid so we can render 
```
grid-template-areas: "header" "content";
```
or so we can change the grid area for `spectrum-InLineAlert-header` to be
```
grid-area: header / icon;
```

Reply:
<img width="819" alt="Screenshot 2023-05-09 at 2 14 28 PM" src="https://github.com/adobe/react-spectrum/assets/698229/21907679-9fd8-4169-8357-1188f37bd4de">

Reply:
that is an interesting idea, neutral is the only one without an icon, we don't let anyone else determine that. I think this would be fine